src/testUtils/errorRoutesApp.ts

```diff
@@ -0,0 +1,48 @@
+import type { RouteOptions, FastifyInstance } from 'fastify';
+import fastifyRoutes from '@fastify/routes';
+
+import type { PluginDone } from '../utilities/fastify/PluginDone.js';
+import type { FastifyTypedInstance } from '../utilities/fastify/FastifyTypedInstance.js';
+
+function registerErrorRoutes(
+  fastify: FastifyTypedInstance,
+  _opts: RouteOptions,
+  done: PluginDone,
+): void {
+  fastify.route({
+    method: ['POST'],
+    url: '/5xx',
+
+    handler(): void {
+      throw new Error('ERROR_MESSAGE');
+    },
+  });
+
+  fastify.route({
+    method: ['POST'],
+    url: '/4xx',
+
+    schema: {
+      body: {
+        type: 'object',
+
+        properties: {
+          test: { type: 'string' },
+        },
+
+        required: ['test'],
+      } as const,
+    },
+
+    handler(): void {
+      this.log.info('SCHEMA_INCORRECTLY_VALIDATED', {});
+    },
+  });
+
+  done();
+}
+
+export async function errorAppPlugin(server: FastifyInstance) {
+  await server.register(fastifyRoutes);
+  await server.register(registerErrorRoutes);
+}
```

Comment:
We could also merge this whole file into the test itself

Reply:
Agreed. This isn't a _utility_ in the sense we use it here, as it isn't (and shouldn't be) shared by multiple test suites.

But more importantly, I don't think we should be defining one plugin for both tests. Unit tests should minimise shared fixtures and that can be easily be accomplished here. For example (untested code, based on the [docs](https://www.fastify.io/docs/latest/Reference/Errors/)):

```ts
import { errorCodes } from 'fastify';

function registerUncaughtErrorRoute(
  fastify: FastifyTypedInstance,
  _opts: RouteOptions,
  done: PluginDone,
): void {
  fastify.route({
    method: ['POST'],
    url: '/uncaught-error',

    handler(): void {
      throw new Error('ERROR_MESSAGE');
    },
  });

  done();
}


function registerClientErrorRoute(
  fastify: FastifyTypedInstance,
  _opts: RouteOptions,
  done: PluginDone,
): void {
  fastify.route({
    method: ['POST'],
    url: '/4xx',

    handler(): void {
      throw new errorCodes.FST_ERR_NOT_FOUND();
      // either that, or the following. Not sure which one should be used here.
      // reply.code(404).send(new errorCodes.FST_ERR_NOT_FOUND())
    },
  });

  done();
}
````

Note how simple the 4XX route can be. I don't think we should be introducing a dependency on something external like `'@fastify/routes'` -- especially when that approach causes unreachable code like the current implementation (the handler is never called).